integration-tests/src/test/no-proving/MicroDao.test.ts

```diff
@@ -0,0 +1,1435 @@
+    advanceOp.verifierKey = TestResource.operationVerifierKey();
+
+    const buyInOp = new ContractOperation();
+    buyInOp.verifierKey = TestResource.operationVerifierKey();
+
+    const cashOutOp = new ContractOperation();
+    cashOutOp.verifierKey = TestResource.operationVerifierKey();
+
+    const setTopicOp = new ContractOperation();
+    setTopicOp.verifierKey = TestResource.operationVerifierKey();
+
+    const voteCommitOp = new ContractOperation();
+    voteCommitOp.verifierKey = TestResource.operationVerifierKey();
+
+    const voteRevealOp = new ContractOperation();
+    voteRevealOp.verifierKey = TestResource.operationVerifierKey();
+
+    return { advanceOp, buyInOp, cashOutOp, setTopicOp, voteCommitOp, voteRevealOp };
+  }
+
+  function part1Deploy({
+    state,
+    orgPk,
+    ops,
+    unbalancedStrictness,
+    balancedStrictness
+  }: {
+    state: TestState;
+    orgPk: Value;
+    ops: ReturnType<typeof setupOperations>;
+    unbalancedStrictness: WellFormedStrictness;
+    balancedStrictness: WellFormedStrictness;
+  }): { addr: ContractAddress; encodedAddr: Uint8Array } {
+    console.log(':: Part 1: Deploy');
+
+    const contract = new ContractState();
+    contract.setOperation(ADVANCE, ops.advanceOp);
+    contract.setOperation(BUY_IN, ops.buyInOp);
+    contract.setOperation(CASH_OUT, ops.cashOutOp);
+    contract.setOperation(SET_TOPIC, ops.setTopicOp);
+    contract.setOperation(VOTE_COMMIT, ops.voteCommitOp);
+    contract.setOperation(VOTE_REVEAL, ops.voteRevealOp);
+
+    contract.data = getChargedState(orgPk);
+    contract.maintenanceAuthority = new ContractMaintenanceAuthority([], 1, 0n);
+
+    const deploy = new ContractDeploy(contract);
+    const tx0 = Transaction.fromParts(
+      LOCAL_TEST_NETWORK_ID,
+      undefined,
+      undefined,
+      testIntents([], [], [deploy], state.time)
+    );
+
+    const addr: ContractAddress = tx0.intents!.get(1)!.actions[0].address;
+    const encodedAddr = encodeContractAddress(addr);
+
+    tx0.wellFormed(state.ledger, unbalancedStrictness, state.time);
+    const balanced = state.balanceTx(tx0.eraseProofs());
+    const proved = balanced.eraseProofs();
+    proved.wellFormed(state.ledger, balancedStrictness, state.time);
+    state.assertApply(proved, balancedStrictness);
+
+    return { addr, encodedAddr };
+  }
+
+  function part2SetTopic({
+    state,
+    addr,
+    orgSk,
+    orgPk,
+    setTopicOp,
+    unbalancedStrictness,
+    balancedStrictness
+  }: {
+    state: TestState;
+    addr: ContractAddress;
+    orgSk: Uint8Array;
+    orgPk: Value;
+    setTopicOp: ContractOperation;
+    unbalancedStrictness: WellFormedStrictness;
+    balancedStrictness: WellFormedStrictness;
+  }) {
+    console.log(':: Part 2: Setting topic');
+
+    const context = new QueryContext(new ChargedState(state.ledger.index(addr)!.data.state), addr);
+    const program = programWithResults(
+      [
+        ...cellRead(getKey(0), false),
```

Comment:
It would be nice to put a comment nearby describing in a human language what this action is about :)